### DISCUSSION TITLE
Jest examples

### DISCUSSION THREAD

**Discussion by andrewnicols** [facebook/docusaurus#7076] [Q&A]
We're considering Docusaurus as a documentation platform our product (Moodle). We're predominantly a PHP platform, but obviously have a fair amount of Javascript too.

I've written a couple of remark plugins, and I'd like to test them with Jest, but I'm hitting some issues in doing so. These remark plugins are not published moidules, but are located in the `src/remark` directory as they're just simple translations.

I'm sure it's something easy/stupid/simple to sort out, but I've been struggling to work out how to address it.

My test is ridiculously simple:
```
const module = require('./trackerLinks');

describe('trackerLinks', () => {
  it('Must exist', () => {
    expect(module).toBeDefined();
});
```

But when I try to run jest, it complains about the use of import statements outside a module:
```
    /Users/nicols/git/moodlehq/dinodevdocs/src/remark/trackerLinks.js:1
    ({"Object.<anonymous>":function(module,exports,require,__dirname,__filename,jest){import _asyncToGenerator from "/Users/nicols/git/moodlehq/dinodevdocs/node_modules/@babel/runtime/helpers/esm/asyncToGenerator.js";
                                                                                      ^^^^^^

    SyntaxError: Cannot use import statement outside a module

    > 1 | const module = require('./trackerLinks');
        | ^
      2 |
      3 | describe('trackerLinks', () => {
      4 |     it('Must exist', () => {

      at Runtime.createScriptFromCode (node_modules/jest-runtime/build/index.js:1728:14)
      at Object.<anonymous> (src/remark/trackerLinks.test.js:1:1)
```

If I modify the `babel.config.js` to use the `@babel/preset-env` then the tests work, but obviously that breaks Docusaurus itself.

I realise that this is not a Docusaurus question or issue as such, but it would be great if there were some documentation to encourage testing things like components and remark plugins within the project.

I note that in the Docusaurus code itself, if I run the jest tests from within the website directory I get a similar error.

Any pointers or guidance would be great.

Thanks,

Andrew

**Comment by slorber** [comment]
Can't tell exactly what's the problem sorry.

That should be possible to use 2 distinct babel configs.


Didn't try but the docs say it could work:

```js
transform: {"\\.[jt]sx?$": ["babel-jest",{configFile: "babel-test.config.js"}]}
```

**Comment by lanegoolsby** [comment]
I was able to get around this issue. See the tsconfig.json file in [this thread](https://github.com/facebook/docusaurus/discussions/7234). I am unable to get past the next error I'm running into though.
